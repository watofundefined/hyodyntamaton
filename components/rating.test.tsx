import { render } from 'test/test-utils'
import Rating from './rating'

describe('Rating component', () => {
  it('matches snapshot for 0', async () => {
    expect(render(<Rating score={0} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 1', async () => {
    expect(render(<Rating score={1} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 2', async () => {
    expect(render(<Rating score={2} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 3', async () => {
    expect(render(<Rating score={3} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 4', async () => {
    expect(render(<Rating score={4} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 5', async () => {
    expect(render(<Rating score={5} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 2.25', async () => {
    expect(render(<Rating score={2.25} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 3.5', async () => {
    expect(render(<Rating score={3.5} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 4.75', async () => {
    expect(render(<Rating score={4.75} />).asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for 4.051', async () => {
    expect(render(<Rating score={4.051} />).asFragment()).toMatchSnapshot()
  })
})
