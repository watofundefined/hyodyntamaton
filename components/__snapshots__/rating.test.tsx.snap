// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component matches snapshot for 0 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 0 out of 5"
    class="rating"
    title="Rating: 0 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(100% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(200% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(300% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(400% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(500% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 1 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 1 out of 5"
    class="rating"
    title="Rating: 1 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(100% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(200% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(300% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(400% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 2 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 2 out of 5"
    class="rating"
    title="Rating: 2 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(100% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(200% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(300% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 2.25 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 2.25 out of 5"
    class="rating"
    title="Rating: 2.25 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(75% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(175% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(275% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 3 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 3 out of 5"
    class="rating"
    title="Rating: 3 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(100% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(200% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 3.5 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 3.5 out of 5"
    class="rating"
    title="Rating: 3.5 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(50% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(150% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 4 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 4 out of 5"
    class="rating"
    title="Rating: 4 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(100% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 4.051 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 4.051 out of 5"
    class="rating"
    title="Rating: 4.051 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(94.89999999999998% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 4.75 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 4.75 out of 5"
    class="rating"
    title="Rating: 4.75 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(25% 0 0 0);"
      />
      <span
        class="circle-outline"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Rating component matches snapshot for 5 1`] = `
<DocumentFragment>
  <span
    aria-label="Rating: 5 out of 5"
    class="rating"
    title="Rating: 5 out of 5"
  >
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
    <span
      aria-hidden="true"
      class="circle-container"
    >
      <span
        class="circle"
        data-testid="circle"
        style="clip-path: inset(0% 0 0 0);"
      />
    </span>
  </span>
</DocumentFragment>
`;
